### PR TITLE
Convert notices to markdown and include CRIU and OpenSSL

### PR DESCRIPTION
**Pull request by pshipton**
Issue https://github.com/eclipse-openj9/openj9/issues/17164

See https://www.eclipse.org/projects/handbook/#legaldoc-notice

### CODE REVIEW

**Review by tajila — APPROVED** [review]

File: NOTICES.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+possession, or use, and re-export of encryption software, to see if this is permitted.
+
+
+## Third-party Content
+
+Checkpoint/Restore In Userspace (CRIU) (3.17.1)
+
+* License: GPL v2, LGPL v2.1 [https://github.com/checkpoint-restore/criu/blob/criu-dev/COPYING](https://github.com/checkpoint-restore/criu/blob/criu-dev/COPYING)
+* Project: https://criu.org/Main_Page
+* Source: https://github.com/ibmruntimes/criu
```
Comment (keithc-ca) [inline NOTICES.md:109]: This doesn't seem right. To my knowledge, we're not including any content from the CRIU project: we're only a client in the same sense that we're a client of Linux (on that platform), so if we don't need to mention Linux, we don't need to mention CRIU.

Comment (pshipton) [reply NOTICES.md:109]: OpenJ9 contains code calling CRIU APIs, see https://www.eclipse.org/projects/handbook/#ip-third-party
Items which may apply are:
- Includes of header files in C/C++
- Use of reflection or other means to reference APIs and implementation

Comment (pshipton) [reply NOTICES.md:109]: See also the part "We cut the dependency graph off at the platform."

Comment (keithc-ca) [reply NOTICES.md:109]: Fair enough, but then we should call out OpenSSL (and probably more).

Comment (pshipton) [reply NOTICES.md:109]: The OpenSSL usage is in the extensions and the notice is in https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/openj9/openj9-openjdk-notices

OMR should likely call out the CUDA Toolkit but I thought we could look at that separately, it's been missing for a long time.

Comment (keithc-ca) [reply NOTICES.md:109]: The handbook says

> We regard the platform as something that the adopter installs and agrees to the licensing terms independently from the project content. ... If a layer of software is installed as part of installing your project’s artifacts, then that layer of software is a dependency.

Users would have to install each of CRIU, CUDA and OpenSSL themselves to enable use by OpenJ9, so my interpretation is that each of those packages should be considered part of the "platform".

The JIT server feature of OpenJ9 uses OpenSSL directly, so if we decide it needs to be referenced, the mention in the extensions is not sufficient.

**Review by keithc-ca — APPROVED** [review]
We'll need changes similar to https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/583 for other versions before this can be merged.

**Pull request merged**